ui/app/components/app/whats-new-popup/whats-new-popup.js

```diff
@@ -0,0 +1,166 @@
+import React, { useContext, useMemo, useRef } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { I18nContext } from '../../../contexts/i18n';
+import { useEqualityCheck } from '../../../hooks/useEqualityCheck';
+import { MetaMetricsContext } from '../../../contexts/metametrics.new';
+import Button from '../../ui/button';
+import Popover from '../../ui/popover';
+import { updateViewedNotifications } from '../../../store/actions';
+import { UI_NOTIFICATIONS } from '../../../../../shared/notifications';
+import {
+  getSortedNotificationsToShow,
+  getSwapsEthToken,
+} from '../../../selectors';
+import { BUILD_QUOTE_ROUTE } from '../../../helpers/constants/routes';
+
+function notifcationActionFunctions(state, metricsEvent) {
+  const swapsEthToken = getSwapsEthToken(state);
+
+  const actionFunctions = {
+    1: () => {
+      metricsEvent({
+        event: 'Swaps Opened',
+        properties: { source: 'Main View', active_currency: 'ETH' },
+        category: 'swaps',
+      });
+      global.platform.openExtensionInBrowser(
+        BUILD_QUOTE_ROUTE,
+        `fromAddress=${swapsEthToken.address}`,
```

Comment:
Isn't this a constant? :confused:. It looks like we're pulling in Redux state here to read a constant that we could instead be importing.

Reply:
That would let us move `actionFunctions` to a module-level constant as well, and avoid re-constructing it each time `notificationActionFunctions` is called. The action functions should be a constant. Any reliance on state can be handled by action creators rather than here where the functions are declared.

Reply:
addressed in 6f6f4026e